codec-socks/src/main/java/io/netty/contrib/handler/codec/socksx/v4/Socks4ServerDecoder.java

```diff
@@ -127,14 +128,15 @@ private void fail(ChannelHandlerContext ctx, Exception cause) {
     /**
      * Reads a variable-length NUL-terminated string as defined in SOCKS4.
      */
-    private static String readString(String fieldName, ByteBuf in) {
-        int length = in.bytesBefore(Math.min(in.readableBytes(), MAX_FIELD_LENGTH + 1), (byte) 0);
+    private static String readString(String fieldName, Buffer in) {
+        int length = in.openCursor(in.readerOffset(), Math.min(in.readableBytes(), MAX_FIELD_LENGTH + 1))
+                .process(ByteProcessor.FIND_NUL);
```

Comment:
We could still use `bytesBefore` and check its less than `MAX_FIELD_LENGTH` afterwards, which would be the common case.